core/local/steps/initial_diff.js

```diff
@@ -93,11 +103,13 @@ async function initialDiff (buffer, out, pouch) {
     }
 
     out.push(batch)
+    events = await buffer.pop()
```

Comment:
`const` are block scoped, using a variable from outside the `while` block may lead to unwanted behavior.

Reply:
`events` is declared outside of the `while` loop (line 26)

Reply:
That's right, that's what I was saying. What I meant was that there is no reason to declare this variable outside the `while' loop.

Reply:
No, I don't. We may be somewhere where the only way to do it is what you did.